sql/src/main/java/io/crate/execution/ddl/views/TransportCreateViewAction.java

```diff
@@ -0,0 +1,120 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.execution.ddl.views;
+
+import io.crate.metadata.TableIdent;
+import io.crate.metadata.view.ViewsMetaData;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.master.TransportMasterNodeAction;
+import org.elasticsearch.cluster.AckedClusterStateUpdateTask;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.block.ClusterBlockException;
+import org.elasticsearch.cluster.block.ClusterBlockLevel;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.metadata.MetaData;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.Priority;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+
+public final class TransportCreateViewAction extends TransportMasterNodeAction<CreateViewRequest, CreateViewResponse> {
+
+    @Inject
+    public TransportCreateViewAction(Settings settings,
+                                     TransportService transportService,
+                                     ClusterService clusterService,
+                                     ThreadPool threadPool,
+                                     ActionFilters actionFilters,
+                                     IndexNameExpressionResolver indexNameExpressionResolver) {
+        super(settings,
+            "crate/sql/views/create",
+            transportService,
+            clusterService,
+            threadPool,
+            actionFilters,
+            indexNameExpressionResolver,
+            CreateViewRequest::new);
+    }
+
+    @Override
+    protected String executor() {
+        return ThreadPool.Names.GENERIC;
+    }
+
+    @Override
+    protected CreateViewResponse newResponse() {
+        return new CreateViewResponse(false);
+    }
+
+    @Override
+    protected void masterOperation(CreateViewRequest request, ClusterState state, ActionListener<CreateViewResponse> listener) {
+        ViewsMetaData views = state.metaData().custom(ViewsMetaData.TYPE);
+        if (conflictsWithTable(request.name(), state.metaData()) || conflictsWithView(request, views)) {
```

Comment:
Isn't the check for "OR REPLACE" flag missing here?

Reply:
ah sorry it's inside the method...

Reply:
Can't replaces tables, even if OR REPLACE is set it's an error

Reply:
yep, yep, sorry I rushed and didn't check the methods used here.